<?xml version="1.0"?>
<launch>

  <!-- start camera -->
  <include file="$(find smocap)/launch/ueye_cam_ceiling_6mm.launch"/>
  
  <!-- spawn marker localization node -->
  <node name="smocap_node" pkg="smocap" type="smocap_node.py">
    <param name="publish_image"   type="bool"   value="true"  />
    <param name="publish_est"     type="bool"   value="true" />
    <param name="camera"          type="string" value="ueye_enac_ceiling_1_6mm"/>
    <param name="detect_rgb"      type="bool"   value="false"  />
    <param name="detect_min_area" type="int"    value="12"    />
  </node>

  <!-- spawn static transform publishers
         - world to camera_link
         - camera_link to camera_optical_frame
         - world to map
  -->
  <node pkg="tf" type="static_transform_publisher" name="world_to_cam_link_publisher"
	args="1 1 2.9 0 1.57079632679 0 world camera_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="cam_link_to_cam_optical_frame_publisher"
	args="0 0 0 -0.5 0.5 -0.5 0.5 camera_link camera_optical_frame 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_map_publisher"
	args="0 0 0 0 0 0 world map 100"/>
  
</launch>

